packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -37,11 +37,12 @@ import recordDataFor from '../record-data-for';
 
 type DefaultRecordData = import('@ember-data/record-data/-private').RecordData;
 type RecordArray = InstanceType<typeof RecordArray>;
+type RelationshipRecordData = import('@ember-data/record-data/-private/ts-interfaces/relationship-record-data').RelationshipRecordData;
 
 // once the presentation logic is moved into the Model package we can make
 // eliminate these lossy and redundant helpers
 function relationshipsFor(instance: any) {
-  let recordData = recordDataFor(instance) || instance;
+  let recordData = (recordDataFor(instance) || instance) as RelationshipRecordData;
```

Comment:
Do we have the ability to type `instance` *here*?

Also, can we add a return type annotation to constrain it (keeping it from breaking in the future)?

Reply:
@chriskrycho I can look into it, would need to be a `import()` to get the type of a class